mava/wrappers/jaxmarl.py

```diff
@@ -271,22 +271,76 @@ def action_spec(self) -> specs.Spec:
         return jaxmarl_space_to_jumanji_spec(merge_space(self._env.action_spaces))
 
     def reward_spec(self) -> specs.Array:
-        return specs.Array(shape=(self._env.num_agents,), dtype=float, name="reward")
+        return specs.Array(shape=(self.num_agents,), dtype=float, name="reward")
 
     def discount_spec(self) -> specs.BoundedArray:
         return specs.BoundedArray(
             shape=(self.num_agents,), dtype=float, minimum=0.0, maximum=1.0, name="discount"
         )
 
+    @abstractmethod
     def action_mask(self, state: JaxMarlState) -> Array:
         """Get action mask for each agent."""
-        if self.has_action_mask:
-            avail_actions = self._env.get_avail_actions(state)
-            mask = jnp.array(batchify(avail_actions, self.agents), dtype=jnp.float32)
-        else:
-            mask = jnp.ones(self._action_shape, dtype=jnp.float32)
-        return mask
+        ...
+
+    @abstractmethod
+    def get_global_state(self, env_state: Array, obs: Dict[str, Array]) -> Array:
+        """Get global state from observation for each agent."""
+        ...
+
+    @property
+    @abstractmethod
+    def state_size(self) -> chex.Array:
+        ...
+
+
+class SmaxWrapper(JaxMarlWrapper):
+    """WWrapper for SMAX environment"""
+
+    def __init__(self, env: MultiAgentEnv, has_global_state: bool = False, timelimit: int = 500):
+        super().__init__(env, has_global_state, timelimit)
+
+    def state_size(self) -> chex.Array:
+        return self._env.state_size
 
-    def get_global_state(self, obs: Dict[str, Array]) -> Array:
+    def action_mask(self, state: JaxMarlState) -> Array:
+        """Get action mask for each agent."""
+        avail_actions = self._env.get_avail_actions(state)
+        return jnp.array(batchify(avail_actions, self.agents), dtype=jnp.float32)
+
+    def get_global_state(self, env_state: Array, obs: Dict[str, Array]) -> Array:
         """Get global state from observation and copy it for each agent."""
-        return jnp.tile(jnp.array(obs["world_state"]), (self._env.num_agents, 1))
+        return jnp.tile(jnp.array(obs["world_state"]), (self.num_agents, 1))
+
+
+class MabraxWrapper(JaxMarlWrapper):
+    """Wrraper for the Mabrax environment."""
+
+    def __init__(self, env: MABraxEnv, has_global_state: bool = False, timelimit: int = 1000):
+        super().__init__(env, has_global_state, timelimit)
+        self.action_shape = self.action_spec().shape
+
+    def state_size(self) -> chex.Array:
```

Comment:
won't work if it's not a property

```suggestion
    @property
    def state_size(self) -> chex.Array:
```